modules/services/syncthing.nix

```diff
@@ -11,6 +11,12 @@ let
 
   isUnixGui = (builtins.substring 0 1 cfg.guiAddress) == "/";
 
+  # syncthing's configuration directory (see https://docs.syncthing.net/users/config.html)
+  syncthing_dir = if pkgs.stdenv.isDarwin then
+    "$HOME/Library/Application Support/Syncthing"
+  else
+    "\${XDG_STATE_HOME:-$HOME/.local/state}/syncthing";
```

Comment:
Quick question. Will the Darwin directory be used even if the XDG_STATE_HOME variable is set?

Reply:
I think so, that's at least how I understood the documentation.

Anyway, I remember that I was also a bit surprised by this when I wrote the module. So just to be sure, this time I took a quick look at the source code (release tag v1.28.0, as used in 24.11). Indeed, `XDG_STATE_HOME` is only considered when [not on Windows/Darwin](https://github.com/syncthing/syncthing/blob/6d64daaba326ba8378c700e00a31e425c7c90579/lib/locations/locations.go#L178-L189) and for Darwin [the location is hardcoded](https://github.com/syncthing/syncthing/blob/6d64daaba326ba8378c700e00a31e425c7c90579/lib/locations/locations.go#L208-L210) as `$HOME/Library/Application Support/Syncthing`.